crates/burn-train/src/metric/base.rs

```diff
@@ -56,6 +59,84 @@ pub trait Metric: Send + Sync {
     fn clear(&mut self);
 }
 
+/// Classification Metric trait
+///
+/// Requires implementation [Metric](Metric)<Input = ClassificationInput<B>>
+pub trait ClassificationMetric<B: Backend>: Metric<Input = ClassificationInput<B>> {
+    /// Sets threshold. Default 0.5
+    fn with_threshold(self, threshold: f64) -> Self;
+    /// Sets average type. Default Micro
+    fn with_average(self, average: ClassificationAverage) -> Self;
+}
+
+/// The [classification metric](ClassificationMetric) input type.
+#[derive(new, Debug)]
+pub struct ClassificationInput<B: Backend> {
+    /// Sample x Class Non thresholded predictions
+    pub predictions: Tensor<B, 2>,
+    /// Sample x Class target mask
+    pub targets: Tensor<B, 2, Bool>,
+}
+
+///Aggregation types for Classification metric average
+#[derive(EnumIter, Copy, Clone, Debug)]
+pub enum ClassificationAverage {
+    /// overall aggregation
+    Micro,
+    /// over class aggregation
+    Macro,
+    // /// over class aggregation, weighted average
+    //Weighted(Box<[f64]>), todo!()
```

Comment:
We can remove the dead code/comment if this is not implemented.

Reply:
Ouki, removed for now